one/src/migrations.rs

```diff
@@ -0,0 +1,157 @@
+use std::path::PathBuf;
+
+use anyhow::{anyhow, Result};
+use async_stream::try_stream;
+use async_trait::async_trait;
+use ceramic_event::unvalidated;
+use ceramic_metrics::{config::Config as MetricsConfig, init_local_tracing};
+use ceramic_service::{Block, BoxedBlock};
+use cid::Cid;
+use clap::{Args, Subcommand};
+use futures::Stream;
+use multihash_codetable::{Code, Multihash, MultihashDigest};
+use tracing::{debug, info};
+
+use crate::{DBOpts, Info};
+
+#[derive(Subcommand, Debug)]
+pub enum EventsCommand {
+    /// Migrate raw event blocks from IPFS.
+    FromIpfs(FromIpfsOpts),
+}
+
+#[derive(Args, Debug)]
+pub struct FromIpfsOpts {
+    /// The path to the ipfs_repo [eg: ~/.ipfs/blocks]
+    #[clap(long, short, value_parser)]
+    input_ipfs_path: PathBuf,
+
+    /// The database to connect to e.g.
+    ///
+    /// `sqlite:///path/to/file.sqlite3` or `sqlite://:memory:` or `sqlite://~/.ceramic-one/db.sqlite3`
+    #[clap(long, short, default_value = "sqlite://~/.ceramic-one/db.sqlite3")]
+    output_database_url: String,
+
+    /// Unique key used to find other Ceramic peers via the DHT
+    #[arg(long, default_value = "testnet-clay", env = "CERAMIC_ONE_NETWORK")]
+    network: crate::Network,
+
+    /// Unique id when the network type is 'local'.
+    #[arg(long, env = "CERAMIC_ONE_LOCAL_NETWORK_ID")]
+    local_network_id: Option<u32>,
+}
+
+impl From<&FromIpfsOpts> for DBOpts {
+    fn from(value: &FromIpfsOpts) -> Self {
+        Self {
+            store_dir: None,
+            database_url: Some(value.output_database_url.clone()),
+        }
+    }
+}
+
+pub async fn events(cmd: EventsCommand) -> Result<()> {
+    if let Err(e) = init_local_tracing() {
+        eprintln!("Failed to initialize tracing: {}", e);
+    }
+    // TODO Cleanup
```

Comment:
either remove or add more detail and link to a ticket?

Reply:
That came over from previous code and since I do not know what it means I'll just delete it.